src/storage/transaction/ChainUpdateEdgeProcessorLocal.cpp

```diff
@@ -0,0 +1,295 @@
+        kvErased_.emplace_back(key);
+        return ECode::SUCCEEDED;
+    }
+
+    auto key = ConsistUtil::edgeKeyPrime(spaceVidLen_, partId, req_.get_edge_key());
+    kvErased_.emplace_back(key);
+
+    std::string val;
+    apache::thrift::CompactSerializer::serialize(req_, &val);
+    val.append(ConsistUtil::updateIdentifier());
+    std::vector<nebula::kvstore::KV> data{{key, val}};
+    auto c = folly::makePromiseContract<ECode>();
+    env_->kvstore_->asyncMultiPut(spaceId_,
+                                  partId,
+                                  std::move(data),
+                                  [p = std::move(c.first)](auto rc) mutable {
+                                      p.setValue(rc);
+                                  });
+    return std::move(c.second);
+}
+
+folly::SemiFuture<ECode> ChainUpdateEdgeProcessorLocal::processRemote(ECode code) {
+    LOG(INFO) << __func__ << "(), code = " << apache::thrift::util::enumNameSafe(code);
+    if (code != ECode::SUCCEEDED) {
+        return code;
+    }
+    auto [pro, fut] = folly::makePromiseContract<ECode>();
+    doRpc(std::move(pro));
+    return std::move(fut);
+}
+
+void ChainUpdateEdgeProcessorLocal::doRpc(folly::Promise<ECode>&& promise,
+                                                     int retry) noexcept {
+    if (retry > retryLimit_) {
+        promise.setValue(ECode::E_LEADER_CHANGED);
+        return;
+    }
+    auto* iClient = env_->txnMan_->getInternalClient();
+    folly::Promise<ECode> p;
+    auto reversedReq  = reverseRequest(req_);
+    iClient->chainUpdateEdge(reversedReq, termOfPrepare_, std::move(p));
+
+    auto f = p.getFuture();
+    std::move(f).thenTry([=, p = std::move(promise)](auto&& t) mutable {
+        auto code = t.hasValue() ? t.value() : ECode::E_RPC_FAILURE;
+        LOG(INFO) << "code = " << apache::thrift::util::enumNameSafe(code);
+        switch (code) {
+            case ECode::E_LEADER_CHANGED:
+                doRpc(std::move(p), ++retry);
+                break;
+            default:
+                p.setValue(code);
+                break;
+        }
+        return code;
+    });
+}
+
+folly::SemiFuture<ECode> ChainUpdateEdgeProcessorLocal::processLocal(ECode code) {
+    LOG(INFO) << __func__ << "(), code = " << apache::thrift::util::enumNameSafe(code);
+    if (code != ECode::SUCCEEDED && code_ == ECode::SUCCEEDED) {
+        code_ = code;
+    }
+    if (processType_ == ChainProcessType::RESUME_REMOTE) {
+        return processResumeRemoteLocal(code);
+    }
+    return processNormalLocal(code);
+}
+
+folly::SemiFuture<ECode>
+ChainUpdateEdgeProcessorLocal::processResumeRemoteLocal(ECode code) {
+    LOG(INFO) << __func__ << "(), code = " << apache::thrift::util::enumNameSafe(code);
+    if (code == ECode::SUCCEEDED) {
+        auto partId = req_.get_part_id();
+        auto key = ConsistUtil::doublePrime(spaceVidLen_, partId, req_.get_edge_key());
+        kvErased_.emplace_back(key);
```

Comment:
Am I missing something here? I think double prime has been add to `kvErased` in `prepareLocal` around line 80

Reply:
good catch